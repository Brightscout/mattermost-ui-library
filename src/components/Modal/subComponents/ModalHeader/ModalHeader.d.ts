export interface ModalHeaderProps {

    /**
     * Main title of the header
     */
    title: string;

    /**
     * Subtitle of the header
     */
    subtitle?: string;

    /**
     * Callback to be triggered on clicking the close icon button
     */
    onHeaderCloseHandler: () => void;
}
