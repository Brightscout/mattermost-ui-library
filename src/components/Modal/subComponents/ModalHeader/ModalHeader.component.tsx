import React from 'react';

import {Icon} from '@Components/Icon';

import {ModalHeaderProps} from './ModalHeader';
import {
    IconButton,
    ModalHeaderRow,
    ModalHeaderSubtitle,
    ModalHeaderTitle,
    ModalHeaderWrapper,
} from './ModalHeader.styles';

/**
 * ModalHeader Component
 *
 * @example
 * ```ts
 * <ModalHeader
 *      title='Modal Header'
        subtitle='Modal Subtitle'
        onCloseHandler={onHideHandler}
 * />
 * ```
 */
export const ModalHeader = (props: ModalHeaderProps) => {
    const {title, subtitle, onCloseHandler} = props;
    return (
        <ModalHeaderWrapper className='mm-modalHeader'>
            <ModalHeaderRow>
                <ModalHeaderTitle className='mm-modalHeaderTitle'>
                    {title}
                </ModalHeaderTitle>
                <IconButton onClick={onCloseHandler}>
                    <Icon
                        name='Close'
                        size={20}
                    />
                </IconButton>
            </ModalHeaderRow>
            {subtitle && (
                <ModalHeaderSubtitle className='mm-modalHeaderSubtitle'>
                    {subtitle}
                </ModalHeaderSubtitle>
            )}
        </ModalHeaderWrapper>
    );
};
