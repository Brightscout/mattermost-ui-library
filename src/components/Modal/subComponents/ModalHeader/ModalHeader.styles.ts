import styled from 'styled-components';

import Colors from '@Styles/colorsForJs.module.scss';

// Styles for the modal header wrapper
export const ModalHeaderWrapper = styled.div({
    padding: '26px  32px',
    borderBottom: `1px solid ${Colors.centerChannel_16}`,
});

// Styles for the modal's first row containing the header, the title and the close icon button
export const ModalHeaderRow = styled.div({
    display: 'flex',
    alignItems: 'center',
    marginBottom: '8px',
});

// Styles for the modal title
export const ModalHeaderTitle = styled.h2({
    margin: 0,
    flex: 1,
    color: Colors.centerChannel,
    fontSize: '22px',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
});

// Styles for the modal subtitle
export const ModalHeaderSubtitle = styled.p({
    margin: 0,
    color: Colors.centerChannel_72,
    fontSize: '12px',
    maxWidth: '80%',
});

// Styles for the icon button
export const IconButton = styled.button({
    background: 'none',
    border: 'none',
});