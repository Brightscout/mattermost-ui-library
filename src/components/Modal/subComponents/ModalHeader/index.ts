export {ModalHeader} from './ModalHeader.component';
