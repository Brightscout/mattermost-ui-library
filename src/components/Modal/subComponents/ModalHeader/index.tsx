import React from 'react';
import {Modal as RBModal} from 'react-bootstrap';

type ModalHeaderProps = {
    title?: string | JSX.Element;
    onHide: () => void;
    showCloseIconInHeader?: boolean;
    className?: string;
}

const ModalHeader = ({
    title,
    showCloseIconInHeader,
    onHide,
    className = '',
}: ModalHeaderProps) : JSX.Element => (
    <>{(title || showCloseIconInHeader) && (
        <RBModal.Header className={`modal__header ${className}`}>
            <div className='modal__title d-flex align-items-center justify-content-between'>
                {title && <p className='modal__title text-ellipsis font-size-20 wt-500 margin-0'>{title}</p>}
                {showCloseIconInHeader && (
                    <button
                        className='style--none'
                        onClick={onHide}
                    >
                        <i className='icon icon-close modal__close-icon cursor-pointer font-size-20'/>
                    </button>
                )}
            </div>
        </RBModal.Header>
    )}
    </>
);

export default ModalHeader;
