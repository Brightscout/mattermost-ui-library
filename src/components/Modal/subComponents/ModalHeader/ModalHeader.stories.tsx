import React from 'react';
import {Story, Meta} from '@storybook/react';

import {ModalHeader} from './ModalHeader.component';
import {ModalHeaderProps} from './ModalHeader';

export default {
    title: 'Component/Modal/ModalHeader',
    component: ModalHeader,
    args: {
        title: 'Modal Header',
    },
} as Meta<typeof ModalHeader>;

const ModalHeaderTemplate: Story<ModalHeaderProps> = (args) => <ModalHeader {...args}/>;

// Default
export const Default = ModalHeaderTemplate.bind({});

export const WithSubtitle = ModalHeaderTemplate.bind({});
WithSubtitle.args = {
    subtitle: 'Diam viverra donec nisi in lorem dictum',
};
