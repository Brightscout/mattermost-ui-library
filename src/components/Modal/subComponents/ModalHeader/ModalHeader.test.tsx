import React from 'react';
import {mount, ReactWrapper} from 'enzyme';

import ModalHeader from '.';

// Mock props data
const hideHandler = jest.fn();
const ModalTitle = 'Modal header';
const ModalHeaderProps = {
    onHide: hideHandler,
    showCloseIconInHeader: false,
    className: 'modal-header-class',
};

describe('Modal Header', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(<ModalHeader {...ModalHeaderProps}/>);
    });

    afterEach(() => {
        component.unmount();
        jest.clearAllMocks();
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed className prop', () => {
        expect(component.hasClass(ModalHeaderProps.className)).toBeTruthy();
    });

    it('Should not render the cross button in the header if the "showCloseIconHeader" prop isn\'t falsy', () => {
        expect(component.find('.icon-close.modal__close-icon').exists()).toBeFalsy();
    });
});

describe('Modal Header - title and close button', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(
            <ModalHeader
                {...ModalHeaderProps}
                title={ModalTitle}
                showCloseIconInHeader={true}
            />,
        );
    });

    afterEach(() => {
        component.unmount();
        jest.clearAllMocks();
    });

    it('Should match snapshot when rendering title and close button also', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should render the title correctly', () => {
        expect(component.text().includes(ModalTitle)).toBeTruthy();
    });

    it('Should render the close button correctly', () => {
        expect(component.find('.icon-close.modal__close-icon').exists()).toBeTruthy();
    });

    it('Should close the modal when close button is clicked', () => {
        component.find('.header-close-btn').simulate('click');
        expect(hideHandler).toHaveBeenCalledTimes(1);
    });
});
