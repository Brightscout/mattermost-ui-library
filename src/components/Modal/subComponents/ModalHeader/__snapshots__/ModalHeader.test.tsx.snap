// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Header - title and close button Should match snapshot when rendering title and close button also 1`] = `ReactWrapper {}`;

exports[`Modal Header Should render correctly and matches snapshot 1`] = `ReactWrapper {}`;
